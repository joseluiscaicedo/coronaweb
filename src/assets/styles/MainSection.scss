@import "./Vars.scss";
.banner{
  background-image: url('https://res.cloudinary.com/josecaicedo/image/upload/v1627427342/coronaweb/Banner_image-COVID_pxtwxz.jpg');
  min-height: 100%;
  background-size: cover;
  background-position-x: center;
  height: max-content;
  display: grid;
  grid-template-columns: minmax(150px, 28%) 1fr;
  color:$main-color;
}

.banner__container{
  background-image: linear-gradient(to right,rgba(255,255,255,0.85) 43%,rgba(255,255,255,0) 100%);
  min-height: 100%;
  background-size: cover;
  background-position-x: center;
  height: max-content;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  padding: 20px 0 20px 28px;
}

.container__firstparagraph{
  padding-top: 2rem;
}