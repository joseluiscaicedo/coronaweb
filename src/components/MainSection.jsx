import React from 'react'
import '../assets/styles/MainSection.scss';

const MainSection = () => {
  return (
      <section className='banner' id='banner' role='banner'>
        <div className='banner__container'>
          <h1>COVID-19 Help Center</h1>
          <p className='container__firstparagraph'>As your healthcare partner, Coronaweb gives you all the information you need about COVID-19 in one place.</p>
          <p>Please login or register to stay updated</p>
        </div>
      </section>
  )
}

export default MainSection;

