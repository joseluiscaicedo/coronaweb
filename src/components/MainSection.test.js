import React from 'react';
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import MainSection from './MainSection';


describe('validate mainSection', () =>{
  test('renders content', () => {
    const mainSection = render(<MainSection />)

    expect(mainSection.length)>0;

  });
  test('Generates Snapshot for mainSection Component', () => {
    const { container } = render(
      <MainSection />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Validate by HeadingElement', () => {
     const component =render( <MainSection /> );
     const HeadingElement=component.getByText('COVID-19 Help Center')
    expect(HeadingElement).toBeInTheDocument();
  });
})