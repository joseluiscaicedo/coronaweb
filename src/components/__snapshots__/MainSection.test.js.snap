// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate mainSection Generates Snapshot for mainSection Component 1`] = `
<section
  class="banner"
  id="banner"
  role="banner"
>
  <div
    class="banner__container"
  >
    <h1>
      COVID-19 Help Center
    </h1>
    <p
      class="container__firstparagraph"
    >
      As your healthcare partner, Coronaweb gives you all the information you need about COVID-19 in one place.
    </p>
    <p>
      Please login or register to stay updated
    </p>
  </div>
</section>
`;
